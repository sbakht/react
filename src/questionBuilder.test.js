// Link.react-test.js
import React from 'react';
import {QuestionBuilder, MaadhiQuestion} from './QuestionBuilder';
import Util from './util';

test('Link changes the class when hovered', () => {
  Util.Root.getRandom = jest.fn();
  Util.Root.getRandom.mockReturnValue({
    root1: 'ك',
    root2: 'س',
    root3: 'ن',
    form: 2,
  });

  const question = new MaadhiQuestion({
    text: 'Which is the 3rd person masculine maadhi?',
    pick: ['past'],
    type: 'type1'
  });

  expect(question.question).toMatchSnapshot();
});
