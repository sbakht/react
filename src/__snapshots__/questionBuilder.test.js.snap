// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
Object {
  "choices": Array [
    Object {
      "isWrong": true,
      "text": Array [
        "كَسَّنَ",
      ],
      "val": Array [
        "كَسَّنَ",
      ],
    },
    Object {
      "isWrong": true,
      "text": Array [
        "كَسَّنَ",
      ],
      "val": Array [
        "كَسَّنَ",
      ],
    },
    Object {
      "isWrong": true,
      "text": Array [
        "كَسَّنَ",
      ],
      "val": Array [
        "كَسَّنَ",
      ],
    },
    Object {
      "isCorrect": true,
      "text": Array [
        Array [
          "ك",
          "َ",
        ],
        Array [
          "س",
          "َ",
        ],
        Array [
          "ن",
          "َ",
        ],
      ],
      "val": Array [
        Array [
          "ك",
          "َ",
        ],
        Array [
          "س",
          "َ",
        ],
        Array [
          "ن",
          "َ",
        ],
      ],
    },
  ],
  "text": "Which is the 3rd person masculine maadhi?",
}
`;
